w1/1_1/main.cpp

```diff
@@ -43,34 +35,38 @@ class DQueue {
     void isBufferFull();
 	void isBufferQuarterEmpty();
     void extendBuffer();
-	void squizeBuffer();
+	void squeezeBuffer();
     void move( int** _pointer );
     
 };
 
-Runner::Runner()
-{
-	queue = DQueue();
-}
+class Runner {
+	
+public:
+	Runner();
+    ~Runner();
+	bool Command( const char& command_id, int& number );
+	
+private:
+	DQueue* _queue;
+};
 
-bool DQueue::Command( const char& command_id, int& number )
-{
-    switch(command_id){
-        case PUSH_BACK:
-            queue.Push_back(number);
-            return 0;
-        case POP_FRONT:
-            return queue.Pop_front() != number;
-    }
-}
+void run( std::istream& inStream, std::ostream& outStream );
+
+// prog.c
 
 DQueue::DQueue()
 {
     //initializing buffer
-    RIGHT = &buffer[ DEFAULT_ARRAY_SIZE - 1 ];
+    buffer = new int[DEFAULT_ARRAY_SIZE];
+    RIGHT = &buffer[DEFAULT_ARRAY_SIZE-1];
     LEFT = &buffer[0];
     
     head = tail = LEFT;
+
+    array_size = DEFAULT_ARRAY_SIZE;
+
+    elementsCount = 0;
 }
 
 DQueue::~DQueue(){};
```

Comment:
Утечка памяти

Reply:
Да, был уверен, что поправил.

```suggestion
DQueue::~DQueue(){
    delete[] LEFT;
};
```